### ISSUE TITLE
Tar dump without cache

### ISSUE THREAD

**Issue by filippocarletti** [restic/restic#3334]
When the cache is missing, restic adds the cache creation message at the beginning of the tar archive, breaking the expected format.
See also https://forum.restic.net/t/tar-dump-without-cache/3682 and https://github.com/NethServer/dev/issues/6456

Output of `restic version`
--------------------------

restic 0.12.0 compiled with go1.15.8 on linux/amd64

How did you run restic exactly?
-------------------------------

```
restic --cache-dir /var/lib/nethserver/backup/restic/rb/ -p /var/lib/nethserver/secrets/restic_rb -r /mnt/backup-rb/ dump latest / | tar -xf - -C /
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------

Local filesystem

Expected behavior
-----------------

Restore files

Actual behavior
---------------

```
tar: This does not look like a tar archive
tar: Skipping to next header
tar: Exiting with failure status due to previous errors
```

Steps to reproduce the behavior
-------------------------------

rm -rf $RESTIC_CACHE_DIR

Do you have any idea what may have caused this?
-----------------------------------------------

When the cache dir is missing, restic adds a message about its creation at the beginning of its output.

Do you have an idea how to solve the issue?
-------------------------------------------

Avoid extra output if dump format is tar or zip.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

restic is helping a lot of happy users.

**Comment by MichaelEischer** [comment]
In case anyone want to work on this: The `created new cache in` message should only be printed if the output is printed on a terminal, which can be detected using `stdoutIsTerminal() `.

**Comment by hluup** [comment]
Created a pull request - https://github.com/restic/restic/pull/3343

**Comment by aawsome** [comment]
I think this issue can be closed after #3343 is merged.
